app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -146,6 +147,21 @@ public void onMainImageTap() {
         openImageInGallery(imageType);
     }
 
+    private boolean copyBarcodeToClipBoard(){
+        if (imageTypes.get(mainImageIndex) == ImageType.BARCODE) {
```

Comment:
This seems to crash if the card has "Barcode Type" set to "No barcode" and has no photo too.

I was about to suggest using the `mainImageIndex != 0` check from `binding.mainImageDescription.setOnClickListener` but I just realized that that check is wrong too (it fails if there is no barcode but an image).

Maybe, given that field does not currently do anything, and copying the text "Front image" or "Back image" is useless... maybe just remove this check altogether and always copy the barcode on long-press?